processing/src/main/java/org/apache/druid/frame/read/columnar/StringFrameColumnReader.java

```diff
@@ -92,18 +85,17 @@ public Column readRACColumn(Frame frame)
     final Memory memory = frame.region(columnNumber);
     validate(memory);
 
-    final long positionOfLengths = getStartOfStringLengthSection(frame.numRows(), false);
-    final long positionOfPayloads = getStartOfStringDataSection(memory, frame.numRows(), false);
-
-    StringFrameColumn frameCol =
-        new StringFrameColumn(
-            frame,
-            false,
-            memory,
-            positionOfLengths,
-            positionOfPayloads,
-            asArray || isMultiValue(memory) // Read MVDs as String arrays
-        );
+    final boolean multiValue = isMultiValue(memory);
+    final long positionOfLengths = getStartOfStringLengthSection(frame.numRows(), multiValue);
+    final long positionOfPayloads = getStartOfStringDataSection(memory, frame.numRows(), multiValue);
+
+    StringFrameColumn frameCol = new StringFrameColumn(
+        frame,
+        multiValue,
+        memory,
+        positionOfLengths,
+        positionOfPayloads
+    );
```

Comment:
is this cool? I sort of remember the rows and columns stuff not handling multi-value strings, which is maybe why it was trying to force it to read as a string array here, though idk that implicit casting like this is good either. Are there any tests that try to feed a multi-value string into window stuff?

Reply:
full disclaimer, i don't actually know what should happen if mvds are used with window stuff, they are super weird and there is no standard to reference so i'm quite unsure what they should do.

Reply:
Currently we dont have any tests on window functions as such operating on the MVDs.
But this case is more of reading data from bytes and if its an MVD, we are setting up the `ColumnType` to `STRING`

FYI, there are tests in `OutputChannelFactoryTest` which were reading MVDs invoking these methods.